Swiften/MIX/MIXImpl.cpp

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2017 Tarun Gupta
+ * Licensed under the simplified BSD license.
+ * See Documentation/Licenses/BSD-simplified.txt for more information.
+ */
+
+#include <Swiften/MIX/MIXImpl.h>
+
+#include <Swiften/Client/StanzaChannel.h>
+#include <Swiften/Elements/IQ.h>
+#include <Swiften/Queries/GenericRequest.h>
+#include <Swiften/Queries/IQRouter.h>
+
+namespace Swift {
+
+MIXImpl::MIXImpl(const JID &ownJID, const JID &channelJID, IQRouter* iqRouter) : ownJID_(ownJID), channelJID_(channelJID), iqRouter_(iqRouter) {
+
+}
+
+MIXImpl::~MIXImpl() {
+
+}
+
+void MIXImpl::joinChannelAndSubscribe(const std::unordered_set<std::string>& nodes) {
+    auto joinPayload = std::make_shared<MIXJoin>();
+    joinPayload->setChannel(channelJID_);
+    for (auto node : nodes) {
+        joinPayload->addSubscription(node);
+    }
+    auto request = std::make_shared<GenericRequest<MIXJoin> >(IQ::Set, getJID(), joinPayload, iqRouter_);
+    request->onResponse.connect(boost::bind(&MIXImpl::handleJoinResponse, this, _1, _2));
+    request->send();
+}
+
+void MIXImpl::joinChannel() {
```

Comment:
I think it would be more useful if this were `joinChannelWithDefaultSubscriptions()`, with the default set of nodes as described at https://xmpp.org/extensions/xep-0369.html#usecase-user-join . Providing the similar features as standard MUC.